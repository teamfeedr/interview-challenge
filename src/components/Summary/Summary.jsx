import React from 'react';
import PropTypes from 'prop-types';

import DietaryIcon from '../DietaryIcon/DietaryIcon';

const getDietaryIcons = ({ dietaryMap }) => {
    return Array.from(dietaryMap).map(([diet, dietCount]) => {
        return (
            <span key={diet}>
                {dietCount}x <DietaryIcon diet={diet} />
            </span>
        )
    })
}

const Summary = ({ selectedCount, dietaryMap }) => {
    const dietaryIcons = getDietaryIcons({ dietaryMap });

    return (
        <div className="menu-summary">
            <div className="container">
                <div className="row">
                    <div className="col-6 menu-summary-left">
                        <span>{selectedCount} items</span>
                    </div>

                    <div className="col-6 menu-summary-right">
                        {dietaryIcons}
                    </div>
                </div>
            </div>
        </div>
    )
}

Summary.propTypes = {
    dietaryMap: PropTypes.instanceOf(Map).isRequired,
    selectedCount: PropTypes.number.isRequired
}

export default Summary;