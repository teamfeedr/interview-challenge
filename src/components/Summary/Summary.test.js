import React from 'react';
import {shallow} from 'enzyme';

import Summary from './Summary';

describe('the Summary component', () => {
    it('renders with dietaryMap and selectedCount', () => {
        const dietaryMap = new Map([
            ['vg', 2],
            ['gf', 1],
            ['v', 65]
        ]);

        const rendered = shallow(<Summary dietaryMap={dietaryMap} selectedCount={3} />);
        
        expect(rendered).toMatchSnapshot();
    })
})