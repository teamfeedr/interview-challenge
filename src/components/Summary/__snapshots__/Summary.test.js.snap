// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Summary component renders with dietaryMap and selectedCount 1`] = `
<div
  className="menu-summary"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-6 menu-summary-left"
      >
        <span>
          3
           items
        </span>
      </div>
      <div
        className="col-6 menu-summary-right"
      >
        <span
          key="vg"
        >
          2
          x 
          <DietaryIcon
            diet="vg"
          />
        </span>
        <span
          key="gf"
        >
          1
          x 
          <DietaryIcon
            diet="gf"
          />
        </span>
        <span
          key="v"
        >
          65
          x 
          <DietaryIcon
            diet="v"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
